soc/riscv32/openisa_rv32m1/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+# Copyright (c) 2018 Foundries.io Ltd
+#
+# SPDX-License-Identifier: Apache-2.0
+
+if(CONFIG_SOC_OPENISA_RV32M1_RI5CY)
+  if(CONFIG_RISCV_GENERIC_TOOLCHAIN)
+    # The RI5CY core has some ISA extensions that are best supported
+    # using the Pulpino toolchain. Use of a generic toolchain is
+    # untested and left to future work.
+    message(FATAL_ERROR "generic toolchain is not supported")
```

Comment:
This assertion should be moved into Kconfig.

Also, it doesn't make sense to have code after a FATAL_ERROR.